experimental/adaptive-cards-2/Catering/CateringDb.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Microsoft.Azure.Cosmos;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Model = Catering.Models;
+
+namespace Catering
+{
+    public class CateringDb : IDisposable
+    {
+        private string _endpointUri;
+        private string _primaryKey;
+
+        // Cosmos client 
+        private CosmosClient _cosmosClient;
+        private Lazy<Database> _database;
+        private Lazy<Container> _container;
+
+        private readonly string DatabaseId = "UserDB";
+        private readonly string ContainerId = "Orders";
+        private readonly string PartitionKeyValue = "AllUsers";
+
+        public CateringDb(IConfiguration configuration)
+        {
+            Configuration = configuration;
+
+            this._endpointUri = Configuration.GetSection("CosmosEndpointUri")?.Value;
+            this._primaryKey = Configuration.GetSection("CosmosKey")?.Value;
+
+            this._cosmosClient = new CosmosClient(_endpointUri, _primaryKey);
```

Comment:
Is this expensive without a HttpClient or HttpClientFactory injected in?

I’m not a C#-primary dev so I’m not sure if this is a good practice or not, but it seems like a potential issue.

Reply:
Yeah, I looked it up and CosmosClient is thread safe so can be injected as a singleton. The queries are not though.